app/controllers/pages_controller.rb

```diff
@@ -10,4 +12,25 @@ def help
       format.html
     end
   end
+
+  private
+
+  def check_uprn_link
+    return if params[:uprn].nil?
+
+    building = Building.find_by(uprn: params[:uprn])
+
+    if building.nil?
+      redirect_to root_path, alert: I18n.t("errors.unknown_uprn", uprn: params[:uprn])
+      return
+    end
+
+    @survey = Survey::Record.find_or_create_by(
+      building_id: building.id,
+      session_id: session.id.to_s,
+      stage: "ownership"
+    )
+
+    redirect_to survey_path(@survey, format: "html")
```

Comment:
Not sure why the `format` is needed but we can use `goto_url` instead.

Reply:
I was getting an UnknownFormatError but I can't seem to reproduce that anymore.

Reply:
it happens in the Cucumber tests, presumably because it's a headless browser?